src/hs_ontology_api/routes/celltypes/celltypes_controller.py

```diff
@@ -0,0 +1,26 @@
+# coding: utf-8
+# JAS November 2023
+from flask import Blueprint, jsonify, current_app, make_response
+from hs_ontology_api.utils.neo4j_logic import celltypedetail_get_logic
+
+
+celltypes_blueprint = Blueprint('celltypes', __name__, url_prefix='/celltypes')
+
+@celltypes_blueprint.route('/<id>', methods=['GET'])
+def celltypes_id_expand_get(id=None):
+    """Returns detailed information on a single cell type.
+
+    :rtype: Union[List[CellTypeDetail]]
+
+    """
+
+    if id =='' or id is None:
```

Comment:
This missing id check will never get executed. When an id is not provided, the call goes to route `/celltypes/` which is not defined at all, and the users will get 404. Remove this unnecessary check.

Reply:
Done. Because this missing id check was also in genes_controller.py and proteins_controller.py, I removed the analogous checks from those files, too.

Reply:
Thanks @AlanSimmons .